rama-http/src/matcher/subdomain_trie.rs

```diff
@@ -0,0 +1,173 @@
+use trie_rs::TrieBuilder;
+use crate::Request;
+use rama_core::{Context, context::Extensions, matcher::Matcher};
+use rama_net::http::RequestContext;
+use rama_net::address::{Host};
+
+#[derive(Debug, Clone)]
+pub struct SubdomainTrieMatcher {
+    trie: trie_rs::Trie<u8>,
+}
+
+impl SubdomainTrieMatcher {
+    pub fn new<I, S>(domains: I) -> Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<str>,
+    {
+        let mut builder = TrieBuilder::new();
+        for domain in domains {
+            let reversed = reverse_domain(domain.as_ref());
+            builder.push(reversed.as_bytes());
+        }
+        let trie = builder.build();
+        Self { trie }
+    }
+
+    pub fn is_match(&self, domain: &str) -> bool {
+        let reversed = reverse_domain(domain);
+        let domain_parts = reversed.split('.');
```

Comment:
my appologies, this is on me, can you please switch to radix_tree crate and use https://docs.rs/radix_trie/0.2.1/radix_trie/struct.Trie.html for your internal trie structure. That one has a nice method called `get_ancestor()` so we can reduce this method to:

```
let reversed = reverse_domain(domain);
self.trie.get_ancestor(&reversed).is_some()
```

seems `trie-rs` is not yet optimised for this use case as it for now collects